tfexec/providers_schema.go

```diff
@@ -2,17 +2,36 @@ package tfexec
 
 import (
 	"context"
+	"fmt"
 	"os/exec"
 
 	tfjson "github.com/hashicorp/terraform-json"
 )
 
+type providersSchemaConfig struct {
+	chdir string
+}
+
+var defaultProvidersSchemaOptions = providersSchemaConfig{}
+
+type ProvidersSchemaOption interface {
+	configureProvidersSchema(*providersSchemaConfig)
+}
+
+func (opt *ChdirOption) configureProvidersSchema(conf *providersSchemaConfig) {
+	conf.chdir = opt.path
+}
+
 // ProvidersSchema represents the terraform providers schema -json subcommand.
 func (tf *Terraform) ProvidersSchema(ctx context.Context) (*tfjson.ProviderSchemas, error) {
```

Comment:
I think in order to actually expose the `chdir` flag here we'd need to introduce `opts ...ProvidersSchemaOption` here as well, no?